src/Razor/src/Microsoft.VisualStudio.RazorExtension/RazorPackage.cs

```diff
@@ -41,9 +41,11 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
                 var componentModel = (IComponentModel)GetGlobalService(typeof(SComponentModel));
                 var breakpointResolver = componentModel.GetService<RazorBreakpointResolver>();
                 var proximityExpressionResolver = componentModel.GetService<RazorProximityExpressionResolver>();
-                var waitDialogFactory = componentModel.GetService<WaitDialogFactory>();
+                var uiThreadOperationExecutor = componentModel.GetService<IUIThreadOperationExecutor>();
                 var editorAdaptersFactory = componentModel.GetService<IVsEditorAdaptersFactoryService>();
-                return new RazorLanguageService(breakpointResolver, proximityExpressionResolver, waitDialogFactory, editorAdaptersFactory);
+                var joinableTaskFactory = new JoinableTaskFactory(ThreadHelper.JoinableTaskContext);
```

Comment:
I believe you can do `ThreadHelper.JoinableTaskContext.Factory`; however, I'd crosscheck that there's not a better way to do this. Typically we import `JoinableTaskContext` in MEF parts so I wonder if the component model / global service have a reference to that (instead of using `ThreadHelper`).

Reply:
Thanks, that's cleaner. I tried doing `componentModel.GetService<JoinableTaskFactory>()` and it didn't have it, but it didn't occur to me to try doing the Context.